hummingbot/hummingbot
# Feat/update ftx spot connector

**Before submitting this PR, please make sure**:


**A description of the changes proposed in the pull request**:
Upgrade of FTX spot connector to the latest standards.
The connector has been transformed in a pure Python connector, and is now a subclass of ExchangePyBase.
All unit tests for it have been implemented.

The PR also includes a change to remove the reference to the configuration map from the AsyncThrottler, that was causing issues during tests runs because the whole Hummingbot connectors configuration was being loaded (and hence the PairFetcher was executed)
It also includes an improvement in the logic that generates client order ids, to reduce the possibility of having repeated ids in different bot instances using the same API keys and trading the same pairs.


**Tests performed by the developer**:
All unit tests are passing in green.


**Tests Summary**:
- Clone and install API keys : ok
- Checking the balance output and cross verifying with exchange : ok
- Create and import strategy : ok
- Start the strategy and run status —live : ok
- Data integrity ( best price, order book, and last price) : ok
- Filled event : ok
- Check history : ok
- Verify fees paid : ok
- Test fast and long refresh rate : ok
- Test multiple levels : ok
- Order cancelation and creation : ok
- Testing hanging orders behaviour : ok
- Flagged issues are resolved

**Tips for QA testing**:
Please perform a standard connector test for FTX spot. Check that all events are being received by the connector (order book is kept correctly synchronized, PNL is calculated correctly for fills and partial feels, etc).


This PR comes from https://github.com/CoinAlpha/hummingbot/pull/278

PRP: https://snapshot.org/#/hbot-prp.eth/proposal/0x7da6ea02f0b14794d94fcd464ad8d4907547067b40f77417683770e2dcd5755c
